### PR TITLE
SDB-2822: Reduce rate window bucket size

### PR DESCRIPTION
**Pull request by fanzhao1986**
Reduce rate window bucket size from second to deci-second.

SDB-2822

### CODE REVIEW

**Review by markaylett — APPROVED** [review]

File: toolbox/net/RateLimit.hpp [hunk 1]
```diff
@@ -57,7 +59,12 @@ TOOLBOX_API std::ostream& operator<<(std::ostream& os, RateLimit rl);
 /// RateWindow maintains a sliding window of second time buckets for the specified interval.
 class TOOLBOX_API RateWindow {
   public:
-    explicit RateWindow(Seconds interval);
+    template <typename intervalT = Seconds>
+    explicit RateWindow(intervalT interval)
+    : buckets_(std::chrono::duration_cast<Decis>(interval).count())
+    {
+        assert(std::chrono::duration_cast<Decis>(interval).count());
```
Comment (Shivix) [inline toolbox/net/RateLimit.hpp:66]: Can you do `assert(buckets_)` here?

File: toolbox/net/RateLimit.cpp [hunk 2]
```diff
@@ -70,10 +65,10 @@ void RateWindow::add(MonoTime time, size_t count) noexcept
     // Clock is assumed to be monotonic, so if the time is not the same, then it must be greater.
     assert(t > last_time_);
     // If the time has advanced by less than one complete cycle of the buffers.
-    if (t - last_time_ < static_cast<time_t>(buckets_.size())) {
+    if ((t - last_time_).count() < int64_t(buckets_.size())) {
```
Comment (Shivix) [inline toolbox/net/RateLimit.cpp:68]:
```suggestion
    if ((t - last_time_).count() < int64_t{buckets_.size()}) {
```

**Review by Shivix — APPROVED** [review]

File: toolbox/net/RateLimit.hpp (hunk 1 above)
Comment (fanzhao1986) [reply toolbox/net/RateLimit.hpp:66]: no. buckets is a vector not an int. but check bucket_.empty() should work.

File: toolbox/net/RateLimit.cpp (hunk 2 above)
Comment (fanzhao1986) [reply toolbox/net/RateLimit.cpp:68]: This is a cast, not initialize. SO it should not work. I need double check.

**Pull request merged**